File: 203 Remove Linked List Elements.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode removeElements(ListNode head, int val) {
        if (head == null) return null;
        ListNode ite = head;
        while (ite.next != null) {            
            if(ite.next.val == val) ite.next = ite.next.next;
            else ite = ite.next;
        }
        if (head.val == val) return head.next;
        else return head;
    }
}

File: 203 Remove Linked List Elements.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def removeElements(self, head: ListNode, val: int) -> ListNode:
        if not head: return head
        dummy = ListNode(0)
        dummy.next = head
        
        pv = dummy
        hd = head
        while hd:
            if hd.val == val:
                hd = hd.next
                pv.next = hd
            else:             
                hd = hd.next   
                pv = pv.next
        
        return dummy.next